File: 392-Is-Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0, j = 0;

        while (i < t.size() && j < s.size()) {
            if (t[i] == s[j])
                i++, j++;
            else
                i++;
        }

        if (j == s.size())
            return true;
        return false;
    }
};

File: 392-Is-Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if not s:
            return True

        i = 0

        for c in t:
            if i < len(s) and c == s[i]:
                i += 1

        if i == len(s):
            return True
        return False        